jira/program.py

```diff
@@ -0,0 +1,72 @@
+"""
+This program listens for Jira events and creates a Google Calendar event.
+
+Scenario:
+    Before completing a feature, a developer is blocked and needs to discuss
+    something with the team.
+
+Workflow:
+    The developer creates a new Jira ticket for the discussion. AutoKitteh
+    automatically generates a Google Calendar event with a deadline for the
+    completion, ensuring that the review happens promptly.
+"""
+
+from datetime import datetime, timedelta
+import os
+
+import autokitteh
+from autokitteh.google import google_calendar_client
+
+
+CALENDAR_CONNECTION_NAME = "my_googlecalendar"
+JIRA_CONNECTION_NAME = "my_jira"
+
+
+def on_jira_issue_created(event):
+    """This function is triggered by an incoming Jira event.
+    It triggers the AutoKitteh workflow to create a Google Calendar event
+    """
+    issue = event.data.issue
+    details = _extract_issue_details(issue)
+    cal = google_calendar_client(CALENDAR_CONNECTION_NAME)
+    _create_calendar_event(cal, details)
+
+
+def _extract_issue_details(issue):
+    base_url = os.getenv(JIRA_CONNECTION_NAME + "__BaseURL")
+    desc = f"Link to Jira Issue: {base_url}/browse/{issue.key}\n\n"
+    issue_details = {
```

Comment:
This will allow you to use `details.xxx` instead of `details["xxx"]` in `_create_calendar_event()`:

```suggestion
    issue_details = autokitteh.AttrDict({
```

Reply:
Done